CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -282,9 +229,67 @@ def add_item(self) -> DB_Response:
     def get_fridge_locations(self):
         pass
 
-class FrigeCheckIn(DB_Item):
-    def __init__(self, db_client: "botocore.client.DynamoDB"):
+
+class FridgeReport(DB_Item):
+
+    REQUIRED_FIELDS = ['fridge_username', 'status', 'notes', 'fridge_percentage']
+    ITEM_TYPES = {
+        'notes': 'S',
+        'fridge_username': 'S',
+        'image_url': 'S',
+        'timestamp': 'N',
+        'status': 'S',
+        'fridge_percentage': 'S'
+    }
+    TABLE_NAME = "fridge_report"
+    VALID_STATUS = set(["working", "needs cleaning", "needs servicing", "not at location"])
+    VALID_FRIDGE_PERCENTAGE = set(["0", "33", "66", "100"])
+    
+    def __init__(self, db_client: 'botocore.client.DynamoDB', fridge_report: dict=None):
         super().__init__(db_client=db_client)
+        if fridge_report is not None:
+            self.notes:str = fridge_report.get('notes', None)
+            self.status:str = fridge_report.get('status', None)
+            self.image_url:str = fridge_report.get('image_url', None)
+            self.fridge_username:str = fridge_report.get('fridge_username', None)
+            self.fridge_percentage:int = fridge_report.get('fridge_percentage', None) 
+    
+    def set_timestamp(self):
+        self.timestamp = str(int(time.time()))
+    
+    def is_valid_status(self) -> bool:
```

Comment:
Like tagname validity checker, shouldn't these methods also be static?

Reply:
Sure, I'll update it :)

Reply:
Done: https://github.com/CollectiveFocus/CFM_Backend/pull/9/commits/d94f5ce519c57b2d19a22bb36d47cd1f9a228867